iree/hal/vulkan/api.h

```diff
@@ -122,6 +183,38 @@ IREE_API_EXPORT iree_status_t IREE_API_CALL
 iree_hal_vulkan_driver_create_default_device(iree_hal_driver_t* driver,
                                              iree_hal_device_t** out_device);
 
+// Creates a Vulkan HAL device using |driver| that shares an existing VkDevice.
+//
+// HAL devices created in this way may share Vulkan resources and synchronize
+// within the same physical VkPhysicalDevice and logical VkDevice directly.
+//
+// |logical_device| is expected to have been created with all extensions
+// returned by |iree_hal_vulkan_get_required_device_extensions| using the
+// options provided during driver creation.
+//
+// The device will schedule commands against the queues in
+// |compute_queues_info| and (if set) |transfer_queues_info|.
+//
+// Applications may choose how these queues are created and selected in order
+// to control how commands submitted by this device are prioritized and
+// scheduled. For example, a low priority queue could be provided to one IREE
+// device for background processing or a high priority queue could be provided
+// for latency-sensitive processing.
+//
+// Dedicated compute queues (no graphics capabilities) are preferred within
+// |compute_queues_info|, if they are available.
+// Similarly, dedicated transfer queues (no compute or graphics) are preferred
+// within |transfer_queues_info|.
+// The queues may be the same.
+//
+// |out_device| must be released by the caller (see |iree_hal_device_release|).
+IREE_API_EXPORT iree_status_t IREE_API_CALL
+iree_hal_vulkan_driver_create_device(
+    iree_hal_driver_t* driver, VkPhysicalDevice physical_device,
+    VkDevice logical_device, iree_hal_vulkan_queues_info_t* compute_queues_info,
```

Comment:
hmm these could be foot-gunny - compute queues are always required, for example, so this feels like an output param (mutable pointer). instead you can make both of these infos value types and then just have queue_count=0 indicate no transfer queues (or, if using a bitfield, == 0). then the only wrong way to call this is with compute_queue count == 0.

Reply:
Made them value types instead of pointers.